core/process.py

```diff
@@ -888,7 +889,32 @@ def Start(self, why):
       raise RuntimeError('Fatal error in posix.fork()')
 
     elif pid == 0:  # child
-      pyos.SignalState_AfterForkingChild()
+      # XXX: If performance is a concern here we may not want to use
+      # `signal_def.GetNumber()` in the calls to `pyos.Sigaction()` below, as
+      # the code generated by `frontend/signal_gen` isn't super efficient
+      # because of the string comparisons.
```

Comment:
This is another important consideration. The performance hit on `fork()` may not be worth the ~10 lines of avoided code duplication...

Reply:
I think you can just use SIGPIPE directly?

In Python you can do

from signal import SIGPIPE

and then in C++, the `leaky_preamble.h` hack should `#include <signal.h>`, which makes SIGPIPE visible

It's a little weird but I think an OK pattern

Reply:
Ah great. That will simplify this